base/channels.jl

```diff
@@ -70,11 +140,110 @@ Closes a channel. An exception is thrown by:
 """
 function close(c::Channel)
     c.state = :closed
-    notify_error(c::Channel, closed_exception())
+    c.excp = Nullable{}(closed_exception())
+    notify_error(c)
     nothing
 end
 isopen(c::Channel) = (c.state == :open)
 
+"""
+    bind(chnl::Channel, task::Task)
+
+Associates the lifetime of `chnl` with a task.
+Channel `chnl` is automatically closed when the task terminates.
+Any uncaught exception in the task is propagated to all waiters on `chnl`.
+
+The `chnl` object can be explicitly closed independent of task termination.
+Terminating tasks have no effect on already closed Channel objects.
+
+When a channel is bound to multiple tasks, the first task to terminate will
+close the channel. When multiple channels are bound to the same task,
+termination of the task will close all channels.
```

Comment:
No it is not obvious, it sounds confusing since it's missing an article or qualifier.